File: app/src/main/java/com/example/commuterx_java/SignupActivity.java
package com.example.commuterx_java;

import static androidx.core.content.ContextCompat.startActivity;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.android.material.button.MaterialButton;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class SignupActivity extends AppCompatActivity {

    FirebaseAuth mAuth;

    /*  @Override <--------------- UNCOMMENT IF U DON'T WANNA LOG IN
    public void onStart() {
        super.onStart();
        FirebaseUser currentUser = mAuth.getCurrentUser();
        if(currentUser != null){
            Intent intent = new Intent(getApplicationContext(), HomeActivity.class);
            startActivity(intent);
            finish();
        }
    } */

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_signup);
        mAuth = FirebaseAuth.getInstance();
        EditText registration_email = findViewById(R.id.registration_email);
        EditText registration_password = findViewById(R.id.registration_password);
        MaterialButton btnRegister = (MaterialButton) findViewById(R.id.btnRegister);

        MaterialButton btnLogin2 = (MaterialButton) findViewById(R.id.btnLogin2);


        btnRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String email, password;
                email = String.valueOf(registration_email.getText());
                password = String.valueOf(registration_password.getText());

                if(TextUtils.isEmpty(email)) {
                    Toast.makeText(SignupActivity.this, "Enter Email", Toast.LENGTH_SHORT).show();
                    return;
                }

                if(TextUtils.isEmpty(password)) {
                    Toast.makeText(SignupActivity.this, "Enter Password", Toast.LENGTH_SHORT).show();
                    return;
                }

                mAuth.createUserWithEmailAndPassword(email, password)
                        .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                            @Override
                            public void onComplete(@NonNull Task<AuthResult> task) {
                                if (task.isSuccessful()) {
                                    Toast.makeText(SignupActivity.this, "Account Created!",
                                            Toast.LENGTH_SHORT).show();
                                    Intent intent = new Intent(SignupActivity.this, MainActivity.class);
                                    startActivity(intent);
                                    finish();

                                } else {
                                    Toast.makeText(SignupActivity.this, "Account Creation Failed.",
                                            Toast.LENGTH_SHORT).show();
                                }
                            }
                        });
            }
        });
    }
}

File: app/src/main/java/com/example/commuterx_java/SignupActivity.kt
package com.example.commuterx_java

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth

class SignupActivity : AppCompatActivity() {
    private lateinit var mAuth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_signup)

        mAuth = FirebaseAuth.getInstance()

        val etEmail = findViewById<EditText>(R.id.etEmail)
        val etPassword = findViewById<EditText>(R.id.etPassword)
        val btnSignUp = findViewById<Button>(R.id.btnRegister)
        val btnGoogle = findViewById<Button>(R.id.btnGoogle)
        val btnFacebook = findViewById<Button>(R.id.btnFacebook)

        btnSignUp.setOnClickListener {
            val email = etEmail.text.toString().trim()
            val password = etPassword.text.toString().trim()

            when {
                TextUtils.isEmpty(email) -> {
                    Toast.makeText(this, "Enter Email", Toast.LENGTH_SHORT).show()
                    return@setOnClickListener
                }
                TextUtils.isEmpty(password) -> {
                    Toast.makeText(this, "Enter Password", Toast.LENGTH_SHORT).show()
                    return@setOnClickListener
                }
                !isPasswordValid(password) -> {
                    Toast.makeText(this,
                        "Password must contain:\n" +
                                "- At least 8 characters\n" +
                                "- One uppercase letter\n" +
                                "- One lowercase letter\n" +
                                "- One number\n" +
                                "- One special character (@#\$%^&+=)",
                        Toast.LENGTH_LONG).show()
                    return@setOnClickListener
                }
                else -> {
                    registerUser(email, password)
                }
            }
        }

        btnGoogle.setOnClickListener {
            // Implement Google Sign-In
            Toast.makeText(this, "Google Sign-In not implemented", Toast.LENGTH_SHORT).show()
        }

        btnFacebook.setOnClickListener {
            // Implement Facebook Sign-In
            Toast.makeText(this, "Facebook Sign-In not implemented", Toast.LENGTH_SHORT).show()
        }
    }

    private fun isPasswordValid(password: String): Boolean {
        val passwordPattern = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=])(?=\\S+$).{8,}$"
        return password.matches(passwordPattern.toRegex())
    }

    private fun registerUser(email: String, password: String) {
        if (!isPasswordValid(password)) {
            Toast.makeText(this,
                "Password must contain at least 8 characters, including uppercase, lowercase, number, and special character",
                Toast.LENGTH_LONG).show()
            return
        }

        mAuth.createUserWithEmailAndPassword(email, password)
            .addOnCompleteListener(this) { task ->
                if (task.isSuccessful) {
                    Toast.makeText(this, "Account Created!", Toast.LENGTH_SHORT).show()
                    startActivity(Intent(this, MainActivity::class.java))
                    finish()
                } else {
                    Toast.makeText(this, "Account Creation Failed: ${task.exception?.message}", Toast.LENGTH_SHORT).show()
                }
            }
    }
}